Frameworks/Foundation/_NSInvocation.arm.asm

```diff
@@ -0,0 +1,87 @@
+;******************************************************************************
+;
+; Copyright (c) Microsoft. All rights reserved.
+;
+; This code is licensed under the MIT License (MIT).
+;
+; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+; THE SOFTWARE.
+;
+;******************************************************************************
+
+    AREA |.text|,ALIGN=4,THUMB,CODE,READONLY
+; r0 = arena start
+; r1 = frame
+; r2 = fn
+; r3 = vfp count
+    ALIGN
+|_CallFrameInternal_VFP| PROC
+    GLOBAL _CallFrameInternal_VFP
+    GLOBAL _CallFrameInternal
+    CMP r3, 1
+    BNE %F0
+    VLDR.64 d0, [r0]
+    B %F1
+
+0
+    VLDM.64 r0, {d0-d7}
+
+1
+    ADD r0, r0, #64
+    ; Fall through to non-VFP case
+|_CallFrameInternal_VFP| ENDP
+|_CallFrameInternal| PROC
+    STM r1, {fp, lr}
+    MOV fp, r1
+
+    MOV sp, r0
+    MOV lr, r2
+    MOV ip, #0
+    LDMIA sp!, {r0-r3}
+    BLX lr
+
+    LDR r2, [fp, #8] ; return pointer
+    LDR r3, [fp, #12] ; return type
+    MOV sp, fp
+
+    ADR ip, %F101
+    TBB [ip, r3]
+
+100
```

Comment:
Minor: the code might be a little clearer with text labels instead of numeric labels. JUMPTABLE, CASE_INT32, etc.